### ISSUE TITLE
[Bug]: Account Settings patches seem to cause hanging

### ISSUE THREAD

**Issue by Nightkingale** [PretendoNetwork/Inkay#32]
### Checked Existing

- [X] I have checked the repository for duplicate issues.

### What happened?

I have not been able to use Inkay when it's set to connect to Pretendo servers. Booting up the console using coldboot Aroma (and isfshax installed) causes the console to hang either when loading the Wii U Menu or soon after. Running Aroma through Health & Safety Information tends to work briefly, but I experience a hang eventually as well.

I did some digging on my end, and reverted all patches relating to Account Settings that were added in the v2.5 release. This seemed to work, as did just using the v2.4 release.

In particular, I commented out all of the following lines in `src/patches/account_settings.cpp` and `src/iosu_url_patches.h`:
https://github.com/PretendoNetwork/Inkay/blob/349b1f38aa2f2ff173d7ebe9af95e919d1ddf50c/src/patches/account_settings.cpp#L125-L133
https://github.com/PretendoNetwork/Inkay/blob/349b1f38aa2f2ff173d7ebe9af95e919d1ddf50c/src/patches/account_settings.cpp#L138-L140
https://github.com/PretendoNetwork/Inkay/blob/349b1f38aa2f2ff173d7ebe9af95e919d1ddf50c/src/iosu_url_patches.h#L38

I'm not exactly sure which lines are causing the issue, as it seems I had to undo all of them in order to get Inkay to work. I don't know what the purpose of these patches are either beyond being used for the upcoming Account Settings stuff, but I fear they might be patching other stuff it shouldn't be as well?

### What did you expect to happen?

The console should load the Wii U Menu successfully without any hanging, and I presume successfully patch Account Settings.

### Steps to reproduce?

I have tried as many troubleshooting tactics as I could think of. So far, I've done the following:
- Formatted my SD card and restored all the files that were on it.
- Redownloaded completely fresh Aroma files to confirm Aroma wasn't the issue.
- Redownloaded the latest version of Inkay from the releases on this repository.
- Isolated Inkay from all other plugins so Inkay was the only plugin I had.

### Other relevant information. (OPTIONAL)

I think it's important to note that I've done a lot of account-related experimentation that a typical user probably shouldn't or wouldn't do, including changing the persistent ID on my user. Without knowing how these patches work, I can't tell if this would be related.

**Comment by ashquarky** [comment]
I agree this looks like an Account Settings issue, though having to make the change to iosu_url_patches is wild - that's a tweak to BOSS/SpotPass, unrelated to this. Without that you probably won't be able to get on Splatoon.

**Comment by ashquarky** [comment]
I've stared at the code and I'm quite confused as to how the account settings patches would cause errors on the *Wii U Menu*. Are you able to source a crash log from `/system_slc/logs`?

**Comment by jonbarrow** [comment]
> that's a tweak to BOSS/SpotPass, unrelated to this

The line he commented out there is for patching the policylist. This *would* line up with the timing of the hang, as the policylist is damn near the first thing the console tries to request during boot. Though like you said, that line is unrelated to the account settings changes...

It sounds like he's having a crash at the same point I was when we were testing out your Hokaku changes, which was caused by us using different firmware versions. I wonder if this is similar..? Though if the same patches work in 2.4.0 I'm not sure why they'd break in 2.5.0...

**Comment by Nightkingale** [comment]
> Are you able to source a crash log from /system_slc/logs?

This is the first place I looked for more information before coming here. Unfortunately, the hanging doesn't produce a crash log. I just now replicated the issue where this time it hung on the "Signing in to Nintendo Network..." message, however my latest crash log is from a few days ago (which was unrelated, Inkay was not present). I'm wondering if it's not actually crashing, just... stuck?

Again, I do want to emphasize the fact that I have done user and account-related tinkering, upwards of changing a user's persistent ID. I think it's very possible it's just something I did on my end which Inkay doesn't play nice with.

If there's another way you suggest me try and get logs or more information I could probably provide, let me know.

**Comment by Nightkingale** [comment]
Hey again! Just wanted to pop in and confirm that the issue is still present for me on Inkay v2.6.0. I bought a serial logging cable to see if I could get anything useful out of it, and I wrote a few logs. The first is cut off for some reason, however the second seems a lot cleaner. I recorded these both during startup, up to when the hanging occurs.

When looking through both though, I honestly don't see how Inkay would be related to this issue. I am wondering if it's an Aroma issue and it's causing Inkay to not want to play nice with my system?

[inkay_troubleshooting_1.log](https://github.com/user-attachments/files/16434356/inkay_troubleshooting_1.log)
[inkay_troubleshooting_2.log](https://github.com/user-attachments/files/16434355/inkay_troubleshooting_2.log)

**Comment by ashquarky** [comment]
Have you rebuilt MLC on this console at all? Other users were seeing strange issues resulting from rebuilding a 5.5.x console into a 5.5.5+ one; since the rebuild doesn't upgrade OSv10.

**Comment by Nightkingale** [comment]
Nope, have not rebuilt MLC on this console. Interesting to note though!

**Comment by Nightkingale** [comment]
Decided to pick up my Wii U again many months later, and it turns out Inkay works now? lol

Literally no idea why, but I'm using the latest version still, and all is well once again. Unless my console did something in standby for the many months it was plugged in and untouched, I'm pretty sure this quite literally fixed itself magically.

I apologize for opening this issue, I'm sure it was just something on my end. Thanks for looking into it regardless.
